Add CI: GitHub Actions for Backend & Frontend Linting, Testing, and Build

AAEmara opened the pull request:
## 📌 Summary

This PR introduces GitHub Actions for continuous integration on the dev branch. It automates backend and frontend checks for every pull request, ensuring code quality, compatibility, and successful builds before merging.

## ✅ What Was Done

<!-- Brief description of major changes -->
  - For backend:
    - [x] Runs a postgres v16 with health checks
    - [x] Uses env variables used by the Django app.
    - [x] Checks on code, sets up Python v3.12, and installs system dependencies
    - [x] Installs Python dependencies using requirements.txt file
    - [x] Installs flake8, black and pytest for linter checks and testing
    - [x] Checks code linting using flake8 and black
    - [x] Tests the Django code using pytest with the current Django settings file.
  - For frontend:
    - [x] Check on code, and sets up Node with v20
    - [x] Install frontend dependencies (node modules)
    - [x] Check code linting using ESLint checks
    - [x] Builds a React app to check on build errors.

## 🧪 How to Test

<!-- How should a reviewer test this locally? Be as specific as possible -->
This workflow runs automatically on pull requests targeting dev.

To test locally:
  - Ensure PostgreSQL is running (or use docker compose up)
  - From backend/, run (install them if they're not available in requirements.txt file):
    ```bash
    pip install -r requirements.txt
    flake8 .
    black --check .
    pytest
    ```


## 🔁 Side Effects / Breaking Changes

<!-- Any potential side effects? DB changes? Backward compatibility issues? -->

- [ ] No breaking changes
- [ ] No database changes
- [ ] No changes to existing endpoint

## 📋 Checklist Before Merge

- [ ] CI workflow was tested via a temporary PR into `feature/github-actions`
- [ ] Linting and tests pass locally (run manually as outlined above)
- [x] Code is clean and self-documented
- [x] Secrets for DB and Django are configured on GitHub
- [ ] Rebased onto latest `dev`

AAEmara committed:
chore(ci): add GitHub Actions for BE & FE workflow

Added GitHub Actions that checks the following:
  - For backend:
    - Runs a postgres v16 with health checks
    - Uses env variables used by the Django app.
    - Checks on code, sets up Python v3.12, and installs system dependencies
    - Installs Python dependencies using requirements.txt file
    - Installs flake8, black and pytest for linter checks and testing
    - Checks code linting using flake8 and black
    - Tests the Django code using pytest with the current Django settings file.

  - For frontend:
    - Check on code, and sets up Node with v20
    - Install frontend dependencies (node modules)
    - Check code linting using ESLint checks
    - Builds a React app to check on build errors.

AAEmara committed:
chore(ci): updating files to fix PR run failure

Fixing the following failures in Dev Branch Checks:
  - React frontend Lint + Build: Run ESLint
    - Erased the commas and added `--ext` between
      each filename.

AAEmara committed:
fix(ci): update dev-checks.yml workflow file

Fixing the GitHub Actions workflow step,
where it included `commas` in the following command:
  - run: npx eslint . --ext .js, .jsx, .ts, .tsx

AhmedRamadanAR commented:
Great Work Abdelrahman

AhmedRamadanAR reviewed (commented):
Great Work Abdelrahaman

AAEmara committed:
fix(ci): update workflow for initial testing

Updating the workflow to bypass flake8 checks.

AAEmara committed:
style(backend): format codebase with Black

Formatted the codebase with Black to pass the
linting checks.

AhmedRamadanAR reviewed (commented):
Great Work Abdelrahman

AAEmara committed:
test(accounts): add dummy test for pytest

Added a dummy test to enable pytest discovery,
and confirm the CI runs the tests properly.

mohamed952741 reviewed (approved):
*Great work, abdelrahman! Incorporating GitHub actions to our work will speed our development and increase our code quality!*
